sources/packages/api/src/esdc-integration/e-cert-full-time-integration/e-cert-full-time-response.service.ts

```diff
@@ -0,0 +1,154 @@
+    filePath: string,
+  ): Promise<ProcessSftpResponseResult> {
+    const result = new ProcessSftpResponseResult();
+    result.processSummary.push(`Processing file ${filePath}.`);
+
+    let responseFile: ECertResponseRecord[];
+
+    try {
+      responseFile = await this.eCertFullTimeService.downloadResponseFile(
+        filePath,
+      );
+    } catch (error) {
+      this.logger.error(error);
+      result.errorsSummary.push(`Error downloading file ${filePath}. ${error}`);
+      // Abort the process nicely not throwing an exception and
+      // allowing other response files to be processed.
+      return result;
+    }
+
+    result.processSummary.push(`File contains ${responseFile.length} records.`);
+
+    for (const feedbackRecord of responseFile) {
+      try {
+        await this.processErrorCodeRecords(feedbackRecord);
+        this.logger.error(
+          `Successfully processed line ${feedbackRecord.lineNumber}.`,
+        );
+      } catch (error) {
+        // Log the error but allow the process to continue.
+        const errorDescription = `Error processing record line number ${feedbackRecord.lineNumber} from file ${filePath}, error: ${error}`;
+        result.errorsSummary.push(errorDescription);
+        this.logger.error(`${errorDescription}. Error: ${error}`);
+      }
+    }
+
+    try {
+      if (result.errorsSummary.length === 0) {
+        // if there is an error in the file do not delete the file
+        await this.eCertFullTimeService.deleteFile(filePath);
+      }
+    } catch (error) {
+      // Log the error but allow the process to continue.
+      // If there was an issue only during the file removal, it will be
+      // processed again and could be deleted in the second attempt.
+      const logMessage = `Error while deleting E-Cert response file: ${filePath}`;
+      this.logger.error(logMessage);
+      result.errorsSummary.push(logMessage);
+    }
+
+    return result;
+  }
+
+  /**
+   * Process the feedback record from the E-Cert response file
+   * and save the error code and disbursementSchedule_id respective to
+   * the document number in DisbursementFeedbackErrors.
+   * @param feedbackRecord E-Cert received record
+   */
+  private async processErrorCodeRecords(
+    feedbackRecord: ECertResponseRecord,
+  ): Promise<void> {
+    const now = getUTCNow();
+    const disbursementSchedule =
+      await this.disbursementScheduleService.getDisbursementScheduleByDocumentNumber(
+        feedbackRecord.documentNumber,
+      );
+    const errorList = [];
+    if (disbursementSchedule) {
+      const updateResult =
+        await this.disbursementScheduleErrorsService.createECertErrorRecord(
+          disbursementSchedule,
+          [
+            feedbackRecord.errorCode1,
+            feedbackRecord.errorCode2,
+            feedbackRecord.errorCode3,
+            feedbackRecord.errorCode4,
+            feedbackRecord.errorCode5,
+          ].filter((error) => error),
```

Comment:
Do we need the _filter_ here? What would be the benefit?

Reply:
as discussed, it for null check